src/astro/databases/base.py

```diff
@@ -103,6 +116,24 @@ def get_table_qualified_name(self, table: Table) -> str:  # skipcq: PYL-R0201
     def default_metadata(self) -> Metadata:
         raise NotImplementedError
 
+    def populate_table_metadata(self, table: Table):
+        # default = self.default_metadata
+        # table_meta = table.metadata
+        # table.metadata = Metadata(
+        #     schema=table_meta.schema or default.schema,
+        #     role=table_meta.role or default.role,
+        #     warehouse=table_meta.warehouse or default.warehouse,
+        #     database=table_meta.database or default.database,
+        #     account=table_meta.account or default.account,
+        #     region=table_meta.region or default.region
+        # )
+
```

Comment:
@kaxil, I'd love your thoughts on this alternative implementation proposed by @dimberman yesterday when we were pairing.

I was slightly worried about mixing values from the connection with the ones defined by the user on the table (IIR we spoke about this, and you were also in favor of an all or nothing approach).

However, Daniel raised an interesting case of a user who may want to only override the `role` for Snowflake and may not have permission to create a connection.

Reply:
Yeah I am. Since the current Airflow users have everything set in their connection, onboarding and even offboarding from Astro SDK is easier if everything is populated from connections